contracts/kill_switch/KillSwitch.sol

```diff
@@ -0,0 +1,130 @@
+pragma solidity 0.4.24;
+
+import "./IKillSwitch.sol";
+import "./IIssuesRegistry.sol";
+import "../apps/AragonApp.sol";
+import "../common/IsContract.sol";
+
+
+contract KillSwitch is IKillSwitch, IsContract, AragonApp {
+    /*
+     * Hardcoded constants to save gas
+     * bytes32 constant public SET_DEFAULT_ISSUES_REGISTRY_ROLE = keccak256("SET_DEFAULT_ISSUES_REGISTRY_ROLE");
+     * bytes32 constant public SET_ISSUES_REGISTRY_ROLE = keccak256("SET_ISSUES_REGISTRY_ROLE");
+     * bytes32 constant public SET_CONTRACT_ACTION_ROLE = keccak256("SET_CONTRACT_ACTION_ROLE");
+     * bytes32 constant public SET_HIGHEST_ALLOWED_SEVERITY_ROLE = keccak256("SET_HIGHEST_ALLOWED_SEVERITY_ROLE");
+     */
+
+    bytes32 constant public SET_DEFAULT_ISSUES_REGISTRY_ROLE = 0xec32b556caaf18ff28362d6b89f3f678177fb74ae2c5c78bfbac6b1dedfa6b43;
+    bytes32 constant public SET_ISSUES_REGISTRY_ROLE = 0xc347b194ad4bc72077d417e05508bb224b4be509950d86cc7756e39a78fb725b;
+    bytes32 constant public SET_CONTRACT_ACTION_ROLE = 0xc7e0b4d70cab2a2679fe330e7c518a6e245cc494b086c284bfeb5f5d03fbe3f6;
+    bytes32 constant public SET_HIGHEST_ALLOWED_SEVERITY_ROLE = 0xca159ccee5d02309b609308bfc70aecedaf2d2023cd19f9c223d8e9875a256ba;
+
+    string constant private ERROR_ISSUES_REGISTRY_NOT_CONTRACT = "KS_ISSUES_REGISTRY_NOT_CONTRACT";
+
+    enum ContractAction { Allow, Check, Deny }
```

Comment:
As we chatted offline, not all actions are useful on a per base/implementation contract basis.

With the current implementation, if checks are enabled for an app when it is upgraded the new implementation contract would need to be enabled for kill switch checks. IMO for most uses, it would be better if the `Check` action is applied per `appId`.

Similarly, the `Allow` action makes more sense to be applied for proxy addresses, which allows to bypass the kill switch check for a specific app. Marking a critical app proxy (like the root authority Voting instance in an organization) with `Allow` would be a way to prevent the organization from ever bricking (at the expense of not being protected by the kill switch)